src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -4,49 +4,93 @@
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int init_capacity = 10;
+    private Object[] data;
+    private int size;
+
+    public <T> ArrayList() {
+        data = new Object[init_capacity];
+        size = 0;
+    }
+
+    private void resizeIfFull() {
+        if (size == data.length) {
+            Object[] newData = new Object[size + 5];
+            System.arraycopy(data, 0, newData, 0, size);
+            data = newData;
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        resizeIfFull();
+        data[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        resizeIfFull();
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
optional:
see how this method is implemented in the source code

Reply:
Yes, that's not optimal realisation, but if we want to implement it like in source code, we need also to implement method toArray() only for one call, because native method System.arrayscopy() recives only arrays type.